source/agora/consensus/data/ValidatorBlockSig.d

```diff
@@ -0,0 +1,60 @@
+/*******************************************************************************
+
+    Contains definition for the `ValidatorBlockSig` struct,
+    which is used to communicate block signatures between nodes.
+
+    Copyright:
+        Copyright (c) 2019-2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.consensus.data.ValidatorBlockSig;
+
+import agora.common.Types;
+import agora.common.crypto.ECC;
+import agora.common.crypto.Key;
+import agora.common.crypto.Schnorr;
+
+/*******************************************************************************
+
+    Define Validator Block Signature information
+
+*******************************************************************************/
+
+public struct ValidatorBlockSig
+{
+    /// The block height of rthis signature
+    public Height height;
+
+    /// The challenge which is the block hash
+    public Scalar challenge;
+
+    /// The public key of the validator
+    public PublicKey public_key;
+
+    /// The block signature for the validator
+    public Signature signature;
```

Comment:
What about signing this entire struct as well? Is it necessary?

Perhaps it's unnecessary because the receiver of this struct will do the schnorr validation check before adding the signature. However, they will still be force to read the block from (cold) storage. So I would recommend adding an additional signature field which signs this entire struct.

There are alternate approaches of course. Node to node encryption is one, and request rate limiting is another. So it's not strictly a requirement to fix this now.